app/src/main/java/com/arctouch/codechallenge/home/HomePresenter.kt

```diff
@@ -0,0 +1,63 @@
+package com.arctouch.codechallenge.home
+
+import android.support.v7.app.AppCompatActivity
+import android.widget.Toast
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.data.MoviesDataSource
+import com.arctouch.codechallenge.data.MoviesRepository
+import com.arctouch.codechallenge.model.Genre
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.util.Injection
+
+class HomePresenter(
+    private val view: HomeContract.View
+) : HomeContract.Presenter,
+    MoviesDataSource.LoadMoviesCallback {
+
+    private var firstLoad = true
+    override val moviesRepository: MoviesRepository = Injection.provideMoviesRepository()
+    override var paginationController = PaginationController(view)
+
+    override fun start() {
+        view.setLoadingIndicator(true)
+        loadGenres()
+        loadMovies(false)
+    }
+
+    private fun loadGenres() {
+        moviesRepository.getGenres(object : MoviesDataSource.LoadGenresCallback {
+            override fun onGenresLoaded(genres: List<Genre>) {
+                // does nothing
```

Comment:
What is the reason for this call?

Reply:
My bad, it should update the cache (fixed in last commit).